bundles/org.openhab.ui/web/src/assets/definitions/blockly/fields/item-field.js

```diff
@@ -18,7 +18,8 @@ export class FieldItemModelPicker extends Blockly.FieldTextInput {
   showEditor_ (options) {
     if (this.f7) {
       const itemsPicked = (value) => {
-        this.setEditorValue_(value.name)
+        this.data = [value.name, value.label]
+        this.setEditorValue_(value.label)
```

Comment:
```suggestion
        this.setEditorValue_(value.name)
```

To have the default behavious of showing the Item also for newly picked Items.